grouper/fe/handlers/permission_disable.py

```diff
@@ -24,11 +21,10 @@ def disable_permission_failed_because_system_permission(self, name):
         # type: (str) -> None
         return self.forbidden()
 
-    def post(self, user_id=None, name=None):
-        repository_factory = RepositoryFactory(self.session, self.graph)
-        service_factory = ServiceFactory(self.session, repository_factory)
-        usecase_factory = UseCaseFactory(service_factory)
-        usecase = usecase_factory.create_disable_permission_usecase(
+    def post(self, *args, **kwargs):
+        # type: (*str, **str) -> None
+        name = kwargs["name"]
```

Comment:
This is interesting. It doesn't look like we use `kwargs` for other handlers. Is this one special, or are we moving in this direction?

Reply:
Yeah, this is because of mypy. It's not a great situation; it's essentially a workaround.

Tornado allows you to define handlers that take arguments, and those arguments are then filled in based on named regex match groups in the route. However, mypy doesn't allow method signatures to vary in method overrides in subclasses, and from its perspective the `post` and `get` methods are method overrides of `RequestHandler`. The method signatures have to match the typeshed entry for `RequestHandler`, which is `(self, *args, **kwargs)`.

This is the workaround that allows these methods to still be typed. Otherwise, we have to either use `# type: ignore` or leave the handler method untyped, both of which are a bit ugly. (Admittedly, this is also ugly.)

Reply:
https://github.com/python/typeshed/issues/1214 is a related issue and recommends `# type: ignore`. This felt marginally less bad than `# type: ignore` to me.